### ISSUE TITLE
overfit when training on small datasets

### ISSUE THREAD

**Issue by Npccc** [keras-team/autokeras#367]
**Feature request**
I found that autokeras often overfit when training on small datasets, so the test accuracy was very low.Can autokeras add some regularization when training?Or, are there other solutions?

**Comment by isaac-gs** [comment]
A few questions,
- How small of a dataset are you talking about?
- Have you created your own network to solve this?
- If so, what was the difference in performance?
- What architecture did autokeras create?

Some datasets are just plain too small. I would recommend looking into data augmentation.

**Comment by Npccc** [comment]
Answer the questions:
1、I extracted five categories from the “Stanford dog” dataset.There are about 100 images in each category
2、I use it finetune pre-training resnet15, and the final classification accuracy is close to 98%.However, the final model generated by autokeras has a classification accuracy of only 78%
3、Autokeras generates a model with nearly 150 layers
4、In constant.py, DATA_AUGMENTATION = True, and whether I need to be augmented while building the dataset??
@isaac-gs thank you

**Comment by isaac-gs** [comment]
Sorry about the delay.

1. That does not seem to be enough. I would look at [this blog post](http://benanne.github.io/2015/03/17/plankton.html) for inspiration on data augmentation techniques. Or many other posts on the topic. The code for this project currently has static data augmentation, but you could pretty easily modify it to do more.
2. Are you saying that your alternative technique was using resnet15 without autokeras? That makes sense. Resnet15 is thoroughly trained to recognize a large number of important features. You should not expect autokeras to be able to beat that with the amount of data you have. For the number of classes in the Stanford dog dataset (unless you aren't using all of them) that seems like pretty good performance.
4. Whether you need to be augmented while building the dataset? What do you mean? You should be able to tell the classifier to do augmentation with an argument. See [this](https://github.com/jhfjhfj1/autokeras/blob/a0efd9b22aae8cf6340977a36ec798a00ae86804/autokeras/image/image_supervised.py), check ImageSupervised for documentation.
